kernel/procfs.c

```diff
@@ -225,6 +245,132 @@ static int _cpuinfo_vcallback(myst_buf_t* vbuf)
     return ret;
 }
 
+#define STATUS_STR "/proc/%d/status"
+
+static int _is_process_traced(char* host_status_buf)
+{
+    assert(host_status_buf);
+
+    char* token;
+    char* save;
+    const char TracerPid[] = "TracerPid:";
+
+    token = strtok_r(host_status_buf, "\n", &save);
+
+    while (token != NULL)
+    {
+        if (strspn(token, TracerPid) == (sizeof(TracerPid) - 1))
+        {
+            char* tracer_pid = token + sizeof(TracerPid) - 1;
+            return atoi(tracer_pid);
+        }
+
+        token = strtok_r(NULL, "\n", &save);
+    }
+    return 0;
+}
+
+static int _status_vcallback(myst_buf_t* vbuf)
+{
+    int ret = 0;
+    struct locals
+    {
+        char status_path[PATH_MAX];
+        myst_thread_t* curr_thread;
+        myst_thread_t* curr_process_thread;
+        char* _host_status_buf;
+    };
+    struct locals* locals = NULL;
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    if (!vbuf)
+        ERAISE(-EINVAL);
+
+    locals->curr_thread = myst_thread_self();
+    locals->curr_process_thread = myst_find_process_thread(locals->curr_thread);
+
+    int n = snprintf(
+        locals->status_path,
+        sizeof(locals->status_path),
+        STATUS_STR,
+        locals->curr_thread->target_tid);
+    if (n >= (int)sizeof(locals->status_path))
+        ERAISE(-ERANGE);
+
+    int size = myst_tcall_get_file_size(locals->status_path);
+
+    if (size <= 0)
+        ERAISE(-EINVAL);
+
+    // allocate extra byte for null termination character.
+    if (!(locals->_host_status_buf = malloc(size + 1)))
+        ERAISE(-ENOMEM);
+
+    ECHECK(myst_tcall_read_file(
+        locals->status_path, locals->_host_status_buf, size));
+    locals->_host_status_buf[size] = 0;
+
+    myst_buf_clear(vbuf);
+    char tmp[128];
+
+    snprintf(tmp, sizeof(tmp), "Name:\t%s\n", locals->curr_thread->name);
```

Comment:
Check the return value to see if there is overflow. Do this for all `snprintf`.

Reply:
Replaced all `snprintf` occurences in this file with `myst_snprintf`. `myst_snprintf` is just a wrapper around the existing snprintf implementation with additional check for overflow.